packages/flutter/test/widgets/fade_in_image_test.dart

```diff
@@ -199,6 +199,46 @@ Future<void> main() async {
       expect(parts.target.opacity, 1);
     });
 
+    // Regression test for https://github.com/flutter/flutter/issues/111011
+    testWidgets("FadeInImage's image obeys gapless playback when first image "
```

Comment:
This test passes even without the fix, can you verify if this test really capture the regression?

Reply:
sorry that it took this long. it took time for me to learn how to "precache" a test image provider. its just that i assumed calling `complete()` before pumping the widget would do the trick but i dove quite deep into `image_test.dart`, just to stumble upon the test below mine which called `resolve()` and then `complete()` which did the trick. i'm not sure i understand the API that clearly, but maybe there's something that can be done to improve it?
anyway, now the test fails on current master. thanks for pointing that out!